app/com/arpnetworking/metrics/portal/alerts/impl/FileSystemAlertRepository.java

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.alerts.impl;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.metrics.portal.alerts.AlertRepository;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import models.internal.AlertQuery;
+import models.internal.Organization;
+import models.internal.QueryResult;
+import models.internal.alerts.Alert;
+import models.internal.impl.DefaultAlert;
+import models.internal.impl.DefaultAlertQuery;
+import models.internal.impl.DefaultMetricsQuery;
+import models.internal.impl.DefaultOrganization;
+import models.internal.impl.DefaultQueryResult;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNegative;
+import net.sf.oval.constraint.NotNull;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Predicate;
+
+/**
+ * An alert repository that loads definitions from the filesystem.
+ *
+ * @apiNote
+ * This repository is read-only, so any additions, deletions, or updates will
+ * result in an {@link UnsupportedOperationException}.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com).
+ */
+public class FileSystemAlertRepository implements AlertRepository {
```

Comment:
On second thought, I'll probably rename this unless it changes considerably from its current form since this name seems to imply that I'm reading each alert individually from their own file vs. all alerts in a single file.